ext/lists.go

```diff
@@ -83,7 +127,35 @@ var comparableTypes = []*cel.Type{
 //	["b", "c", "a"].sort() // return ["a", "b", "c"]
 //	[1, "b"].sort() // error
 //	[[1, 2, 3]].sort() // error
-
+//
+// # SortBy
+//
+// Sorts a list by a key value, i.e., the order is determined by the result of
+// an expression applied to each element of the list.
+//
+// Examples:
+//
+//	[
+//	  Player { name: "foo", score: 0 },
+//	  Player { name: "bar", score: -10 },
+//	  Player { name: "baz", score: 1000 },
+//	].sortBy(e, e.score).map(e, e.name)
+//	== ["bar", "foo", "baz"]
+//
+// # SortByAssociatedKeys
```

Comment:
Could you help me understand the use case for this? This seems like an implementation detail of the `sortBy` macro.

Reply:
I think we can have it as an implementation detail for now (prefixed with @ to show that the function is internal), and if there is a need for it later on we can always expose it.